packages/datadog-plugin-mongodb-core/src/index.js

```diff
@@ -1,14 +1,29 @@
 'use strict'
 
 const DatabasePlugin = require('../../dd-trace/src/plugins/database')
+const coalesce = require('koalas')
 
 class MongodbCorePlugin extends DatabasePlugin {
   static get id () { return 'mongodb-core' }
   static get component () { return 'mongodb' }
   // avoid using db.name for peer.service since it includes the collection name
   // should be removed if one day this will be fixed
   static get peerServicePrecursors () { return [] }
+
+  configure (config) {
+    super.configure(config)
+    this.config.mongoHeartbeatDisabled = coalesce(
+      config.mongoHeartbeatDisabled,
+      process.env.DD_TRACE_MONGO_HEARTBEAT_DISABLED,
+      false
+    )
+  }
+
   start ({ ns, ops, options = {}, name }) {
+    // heartbeat commands can be disabled if this.config.mongoHeartbeatDisabled is true
+    if (isHeartbeatDisabled(ops, this.config)) {
```

Comment:
The name is confusing as it returns if it's a heartbeat, not if it's disabled. Or I guess it checks both which makes naming more difficult. Maybe something like `!config.heartbeatEnabled && !isHeartbeat(ops, this.config` or combine them in some sort of `shouldTrace(ops, this.config)`.

Reply:
understood, fixed the naming